File: api/src/main/java/net/asch/bulkit/api/capability/Capabilities.java
package net.asch.bulkit.api.capability;

import net.asch.bulkit.api.BulkIt;
import net.minecraft.core.Direction;
import net.neoforged.neoforge.capabilities.BlockCapability;
import net.neoforged.neoforge.capabilities.ItemCapability;
import net.neoforged.neoforge.items.IItemHandler;
import org.jetbrains.annotations.Nullable;

public class Capabilities {
    public static class Disk {
        public static ItemCapability<IDiskResourceHandler, @Nullable Void> RESOURCE = ItemCapability.createVoid(BulkIt.location("disk_resource"), IDiskResourceHandler.class);
        public static ItemCapability<IItemHandler, @Nullable Void> MODS = ItemCapability.createVoid(BulkIt.location("disk_mods"), IItemHandler.class);

        public static ItemCapability<IDiskResourceRenderer, @Nullable Void> RESOURCE_RENDERER = ItemCapability.createVoid(BulkIt.location("disk_resource_renderer"), IDiskResourceRenderer.class);
    }

    public static class DriveNetwork {
        public static BlockCapability<IDriveNetworkLink, @Nullable Direction> LINK = BlockCapability.createSided(BulkIt.location("drive_network_link"), IDriveNetworkLink.class);
    }
}

File: api/src/main/kotlin/net/asch/bulkit/api/capability/Capabilities.kt
package net.asch.bulkit.api.capability

import net.asch.bulkit.api.BulkItApi
import net.minecraft.core.Direction
import net.neoforged.neoforge.capabilities.BlockCapability
import net.neoforged.neoforge.capabilities.ItemCapability
import net.neoforged.neoforge.items.IItemHandler

object Capabilities {
    object Disk {
        val RESOURCE: ItemCapability<IDiskResourceHandler, Void?> = ItemCapability.createVoid(
            BulkItApi.location("disk_resource"), IDiskResourceHandler::class.java
        )
        val MODS: ItemCapability<IItemHandler, Void?> = ItemCapability.createVoid(
            BulkItApi.location("disk_mods"), IItemHandler::class.java
        )
    }

    object DriveNetwork {
        var LINK: BlockCapability<IDriveNetworkLink, Direction?> = BlockCapability.createSided(
            BulkItApi.location("drive_network_link"), IDriveNetworkLink::class.java
        )
    }
}